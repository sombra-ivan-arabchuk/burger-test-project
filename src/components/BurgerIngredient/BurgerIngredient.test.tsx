import React from 'react';
import { render, RenderResult, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import BurgerIngredient from './BurgerIngredient';

export interface SetupProps {
  container: HTMLElement;
  utils: RenderResult;
}

const setup = (types: Array<string>): SetupProps => {
  // eslint-disable-next-line no-undef
  const utils = render(
    <div>
      {types.map(type => (
        <BurgerIngredient type={type} />
      ))}
    </div>,
  );
  return {
    container: utils.container,
    utils,
  };
};

describe('Catalog', () => {
  it('check if component renders correct type of ingredient', async () => {
    const { container } = setup(['meat', 'salad']);

    const { firstChild } = container;
    const [meat, salad] = firstChild.children;
    // meat background color
    expect(meat).toHaveStyle('background: linear-gradient(#7f3608, #702e05)');
    // salad background color
    expect(salad).toHaveStyle('background: linear-gradient(#228c1d, #91ce50)');
  });
});
