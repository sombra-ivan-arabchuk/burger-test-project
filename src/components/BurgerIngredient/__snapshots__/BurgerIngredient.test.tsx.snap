// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders 1`] = `
<div>
  <div
    style="width: 80%; height: 8%; margin: 2% auto; border-radius: 15px;"
  />
</div>
`;
